libbeat/monitoring/inputmon/httphandler.go

```diff
@@ -73,10 +73,17 @@ func (h *handler) allInputs(w http.ResponseWriter, req *http.Request) {
 			continue
 		}
 
-		if requestedType != "" {
-			if typ, ok := m["input"].(string); ok && !strings.EqualFold(typ, requestedType) {
-				continue
-			}
+		// Require all entries to have an 'input' and 'id' to be accessed through this API.
+		inputType, ok := m["input"].(string)
+		if !ok {
+			continue
+		}
+		if _, ok = m["id"].(string); !ok {
```

Comment:
Is `m.id` ever non-`string`?

Reply:
I don't expect it to be registered that way (and it won't be if the `inputmon` package is used), but I don't think I can prevent libbeat users from registering an `id` metric that's not a string.